src/Symfony/Component/Mime/NamedAddress.php

```diff
@@ -39,4 +50,17 @@ public function toString(): string
     {
         return $this->getEncodedNamedAddress();
     }
+
+    public static function fromString(string $string): self
+    {
+        if (false === strpos($string, '<')) {
+            return new self($string, '');
```

Comment:
What about returning a simple `Address` instance here instead?

Reply:
I am against this, because in this case `fromString()` would have 2 return types NamedAddress|Address 👎

Reply:
But returning a `NamedAddress` is a lie, as there is no name. And the return type should indeed be changed, but to `Address` only (as `NamedAddress` is a subclass of `Address`). What we care about is having an `Address` in your code (the fact that it can be a `NamedAddress` is an implementation details).

Another possibility would be to remove the distinction between `Address` and `NamedAddress` and only have `Address` which would support named addresses.

Reply:
I also did not like having this method returning two kinds. It could also result in an `InvalidArgumentException` as far as I'm concerned.

Reply:
I think throwing an exception is not an option as the use case is configuration, where you don't really know what the user is going to use.

Reply:
But... 
```php
$address = new NamedAddress('example@example.com', '');
```
is allowed at the moment, maybe that should also not be the case?

Reply:
> Another possibility would be to remove the distinction between Address and NamedAddress and only have Address which would support named addresses.

👍

Reply:
> > Another possibility would be to remove the distinction between Address and NamedAddress and only have Address which would support named addresses.

Would that be a think to do in this PR, or create a separate PR for?

Reply:
> What we care about is having an Address in your code (the fact that it can be a NamedAddress is an implementation details).

This is true, but named constructors are not meant to return anything else than the static class itself as AFAIK. It would be strange to me:
```php
public function myMethod(NamedAddress $foo) {
    // ...
}
```
and get an exception when calling it like this, because its not a NamedAdress:
```php
$this->myMethod(NamedAddress::fromString(.....))
```

Reply:
I have done the PR in #33270 as I wanted to understand why I added `NamedAddress` in the first place and the consequences of its removal. It's safe, so let's do it! when #33270 is merged, we will be able to rebase here and the discussion would be close :)